src/src/com/microsoft/aad/adal/ADALError.java

```diff
@@ -515,8 +515,28 @@
     /**
      * Resource authentication challenge failure
      */
-    RESOURCE_AUTHENTICATION_CHALLENGE_FAILURE("Resource authentication challenge failure");
+    RESOURCE_AUTHENTICATION_CHALLENGE_FAILURE("Resource authentication challenge failure"),
 
+    /**
+     * Export of FID failure
+     */
+    FAIL_TO_EXPORT("Fail to export"),
+    
+    /**
+     * Import of FID failure
+     */
+    FAIL_TO_IMPORT("Fail to import"),
+        
+    /**
+     * Deserialization of FID failure
+     */
+    FAIL_TO_DESERIALIZE("Fail to deserialize"), 
```

Comment:
I don't understand the comment here, Deserialization of FID failure? Is it a family client id deserialization failed or? If so, why we deserialize FID? Shouldn't we deserialize the blob or tokenCacheItem? Sorry, I'm just confused by the comment here, and don't know when we throw the failure.